Mathlib/RingTheory/NormalClosure.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2025 Xavier Roblot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Xavier Roblot
+-/
+import Mathlib.RingTheory.DedekindDomain.IntegralClosure
+
+/-!
+# Normal closure of an extension of domains
+
+We define the normal closure of an extension of domains `R ⊆ S` as a domain `T` such that
+`R ⊆ S ⊆ T` and the extension `Frac T / Frac R` is Galois, and prove several instances about it.
+
+Under the hood, `T` is defined as the `integralClosure` of `S` inside the
+`IntermediateField.normalClosure` of the extension `Frac S / Frac R` inside the `AlgebraicClosure`
+of `Frac S`. In particular, if `S` is a Dedekind domain, then `T` is also a Dedekind domain.
+
+# Technical notes
+* Many instances are proved about the `IntermediateField.normalClosure` of the extension
+`Frac S / Frac R` inside the `AlgebraicClosure` of `Frac S`. However these are only needed for the
+construction of `T` and to prove some results about it. Therefore, these instances are scoped.
+* `Ring.NormalClosure` is defined as a type rather than a `Subalgebra` for performance reasons
+(and thus we need to provide explicit instances for it). Although, defining at a `Subalgebra`
+does not cause timeouts in this file, it does slow down considerably its compilation and
+does trigger timeouts in applications.
+-/
+
+namespace Ring
+
+noncomputable section normalClosure
+
+variable (R S : Type*) [CommRing R] [CommRing S] [IsDomain R] [IsDomain S]
+  [Algebra R S] [NoZeroSMulDivisors R S]
+
+/--
+We register this specific instance as a local instance rather than making
+`FractionRing.listAlgebra` a local instance because the latter causes timeouts since
```

Comment:
`FractionRing.liftAlgebra` a local instance because the latter causes timeouts since